### ISSUE TITLE
Pick & Omit are not working as expected

### ISSUE THREAD

**Issue by Mahmoud-Masri** [typescript-rtti/typescript-rtti#89]
Pick & Omit are not working as expected

```typescript
interface Obj {
    a: string
    b: string
    c: string
}

type ModifiedObj = Omit<Obj, 'a'>
type ModifiedObj2 = Pick<Obj, 'a' | 'b'>

reflect<Obj>().matchesValue({ b: 'value', c: 'value' }) // == false as expected
reflect<ModifiedObj>().matchesValue({ b: 'value', c: 'value' }) // == true as expected
reflect<ModifiedObj>().matchesValue({ c: 'value' }) // == true; expected false, b is required but missing
reflect<ModifiedObj2>().matchesValue({ c: 'value' }) // == true; expected false, a and b are required but missing and c shouldn't be accepted
reflect<ModifiedObj2>().matchesValue({ a: 'value' }) // == true; expected false, b is required but missing
```

```json
"typescript-rtti": "0.8.2",
"reflect-metadata": "0.1.13",
"typescript": "4.8.1-rc",
"ttypescript": "1.5.13"
 ```

**Comment by rezonant** [comment]
Support for this will be in vNext (7a465ab8e3a02ea91997452d76cd68a05cc55587)

**Comment by rezonant** [comment]
Released in 0.8.3

**Comment by rezonant** [comment]
Here's the above example in the playground: https://typescript-rtti.org/#fFYyfHsic291cmNlIjoiaW1wb3J0IHsgcmVmbGVjdCB9IGZyb20gJ3R5cGVzY3JpcHQtcnR0aSc7XG5pbnRlcmZhY2UgT2JqIHtcbiAgICBhOiBzdHJpbmdcbiAgICBiOiBzdHJpbmdcbiAgICBjOiBzdHJpbmdcbn1cblxudHlwZSBNb2RpZmllZE9iaiA9IE9taXQ8T2JqLCAnYSc+XG50eXBlIE1vZGlmaWVkT2JqMiA9IFBpY2s8T2JqLCAnYScgfCAnYic+XG5cbmNvbnNvbGUubG9nKHJlZmxlY3Q8T2JqPigpLm1hdGNoZXNWYWx1ZSh7IGI6ICd2YWx1ZScsIGM6ICd2YWx1ZScgfSkpOyAvLyA9PSBmYWxzZSBhcyBleHBlY3RlZFxuY29uc29sZS5sb2cocmVmbGVjdDxNb2RpZmllZE9iaj4oKS5tYXRjaGVzVmFsdWUoeyBiOiAndmFsdWUnLCBjOiAndmFsdWUnIH0pKTsgLy8gPT0gdHJ1ZSBhcyBleHBlY3RlZFxuY29uc29sZS5sb2cocmVmbGVjdDxNb2RpZmllZE9iaj4oKS5tYXRjaGVzVmFsdWUoeyBjOiAndmFsdWUnIH0pKTsgLy8gPT0gdHJ1ZTsgZXhwZWN0ZWQgZmFsc2UsIGIgaXMgcmVxdWlyZWQgYnV0IG1pc3NpbmdcbmNvbnNvbGUubG9nKHJlZmxlY3Q8TW9kaWZpZWRPYmoyPigpLm1hdGNoZXNWYWx1ZSh7IGM6ICd2YWx1ZScgfSkpOyAvLyA9PSB0cnVlOyBleHBlY3RlZCBmYWxzZSwgYSBhbmQgYiBhcmUgcmVxdWlyZWQgYnV0IG1pc3NpbmcgYW5kIGMgc2hvdWxkbid0IGJlIGFjY2VwdGVkXG5jb25zb2xlLmxvZyhyZWZsZWN0PE1vZGlmaWVkT2JqMj4oKS5tYXRjaGVzVmFsdWUoeyBhOiAndmFsdWUnIH0pKTsgLy8gPT0gdHJ1ZTsgZXhwZWN0ZWQgZmFsc2UsIGIgaXMgcmVxdWlyZWQgYnV0IG1pc3NpbmciLCJjb21waWxlck9wdGlvbnMiOiIifQ==
